_posts/blog/2016-07-07-reactive_mindset_burgers.md

```diff
@@ -0,0 +1,165 @@
+
+Mickey Mouse knows how to handle three situations:
+
+* ``onNext(Burger burger)`` - what to do when he gets a burger.
+* ``onError(Throwable error)`` - what to do in case of an error, for example, when the burger was burned.
+* ``onComplete()`` - what to do when the nephews have gone home for the day.
+
+In code, this would look like this:
+{% highlight java %}
+burgerObservable.subscribe(
+		        burger -> eatBurger(burger), //onNext
+		        error -> complain(error), //onError
+		        () -> leaveBurgerJoint()); //onComplete
+{% endhighlight %}
+Every time Mickey Mouse gets a new burger, he will eat it. Whenever something bad happened with the burger, Mickey will complain. When there are no more burgers to be eaten, Mickey Mouse will leave the burger joint.
+
+## Dining Area vs. Kitchen
+Overall, the burger making and consuming would look, in RxJava, like this:
+
+{% highlight java %}
+Observable.zip(bunObservable, meatObservable, tomatoObservable,
+     	          (bun, meat, tomato) -> makeBurger(bun, meat, tomato))
+	  .subscribe(burger -> eatBurger(burger),
+		    error -> complain(error));
+{% endhighlight %}
+
+Mickey Mouse orders the burger in the dining area of the restaurant. By default, in ReactiveX, the event stream would be processed on the same thread. So, this means that the preparation of the burger would also happen in the dining area. But this should actually happen in the kitchen! This also allows Donald Duck to make more burgers while Mickey Mouse eats his burger.
+ReactiveX provides two methods that allow defining on which thread the operations should happen and on this thread the result should be handled: ``observeOn`` and ``subscribeOn``.
+Donald Duck should make the burger in the kitchen. This will be defined: ``subscribeOn(kitchen)``. Mickey Mouse eats the burger in the dining area: ``observeOn(dining area)``.
+The "kitchen" should be a background thread and the "dining area" should be the main thread.
+
+{% highlight java %}
+Observable.zip(bunObservable, meatObservable, tomatoObservable,
+     	        (bun, meat, tomato) -> makeBurger(bun, meat, tomato))
+	  .subscribeOn(Schedulers.computation())
+	  .observeOn(AndroidSchedulers.mainThread())
+	  .subscribe(burger -> eatBurger(burger),
+		    error -> complain(error));
+{% endhighlight %}
+
+## Reactive Burger Conclusion
+
+Replace the burger joint with your own context; Huey, Dewey, and Louie with your own data sources; the buns, the meat and the tomato slices with your own model data. The concepts are still the same - streams of data that are fairly easy to manipulate and to compose, at the same time, being able to handle the working threads. The ReactiveX implementations of reactive programming in Java, JavaScript, Swift or .NET work great for both backend and frontend and offer a paradigm that can be used when programming any event-driven software.  
```

Comment:
Florina is talking about languages - so in this case Android is not needed. But in this case change .NET to C#